service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/StoragetransferConfiguration.java

```diff
@@ -0,0 +1,44 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.bucket;
+
+import com.google.api.client.googleapis.util.Utils;
+import com.google.api.services.storagetransfer.v1.Storagetransfer;
+import com.google.api.services.storagetransfer.v1.StoragetransferScopes;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class StoragetransferConfiguration {
+  private static final Logger logger = LoggerFactory.getLogger(StoragetransferConfiguration.class);
+
+  @Bean
+  public GoogleCredentials getGoogleCredentials() {
```

Comment:
What is the advantage of having this be a Bean?
What was broken about the previous implementation?
Could we use a "normal" singleton that makes the object on first reference? It would properly barf if called during unit tests.

Reply:
I made `Storagetransfer` into a bean so I could mock it out in the unit test. It depends on this `GoogleCredentials` object in a non-springy way, so I made it a bean too. You're right that it just kicks the can down the road a bit in terms of failing to load the ADCs, but at least now it fails to create a bean with the name of the thing it can't find. ✌🏻 

I'm not sure what a normal singleton is in this context. A spring Component is the usual way to do that, right?

Reply:
You don't need something to be a bean to be able to mock. You can mock any class or interface.
One way to do this is to make a @Component that has a (non-static) method that does a getOrCreate of the Storagetransfer object. Then in your test, mock that method.

Alternately, you could make a non-Spring static class that did the getOrCreate. I think you could use `MockedStatic` do mock it, but I have never used that myself so don't know if it is weird.

Reply:
Haven't tried `MockedStatic`. Your first method sounds like a way to make a bean with more steps, since Components are beans too.

Reply:
The difference is that the bean gets created all the time. The indirection through a method only happens when we need to real transfer.

Reply:
Split the bean via profiles.